admin/controllers/pupil.js

```diff
@@ -95,27 +82,142 @@ const postAddPupil = async (req, res, next) => {
   } catch (error) {
     next(error)
   }
-  res.redirect(`/school/pupil-register/lastName/true?hl=${pupil._id}`)
+  const pupilId = JSON.stringify([pupil._id])
+  res.redirect(`/school/pupil-register/lastName/true?hl=${pupilId}`)
 }
 
-const getAddMultiplePupils = async (req, res, next) => {
+const getAddMultiplePupils = (req, res, next) => {
   res.locals.pageTitle = 'Add multiple pupils'
+  const { hasError } = res
   try {
     req.breadcrumbs('Pupil Register', '/school/pupil-register/lastName/true')
     req.breadcrumbs(res.locals.pageTitle)
     res.render('school/add-multiple-pupils', {
-      breadcrumbs: req.breadcrumbs()
+      breadcrumbs: req.breadcrumbs(),
+      hasError
     })
   } catch (error) {
     next(error)
   }
 }
 
+const postAddMultiplePupils = async (req, res, next) => {
+  let school
+  try {
+    school = await School.findOne({_id: req.user.School}).exec()
+    if (!school) {
+      throw new Error(`School [${req.body.school}] not found`)
```

Comment:
I don't understand why `req.body.school` is used in the error message but `req.user.School` is used for the actual find?

Reply:
its wrong, update coming